src/builder/commands/ubuntu.rs

```diff
@@ -675,57 +675,38 @@ impl Distro {
     }
 
     #[cfg(feature="containers")]
-    fn ensure_eat_my_data(&mut self, ctx: &mut Context)
+    fn ensure_basic_packages(&mut self, ctx: &mut Context, extra_packages: &[&str])
         -> Result<(), StepError>
     {
         if self.eatmydata == EatMyData::Need {
             self.ensure_codename(ctx)?;
-            let eatmy = EMDParams::find(
-                self.codename.as_ref().unwrap(), &self.config.arch);
-            if let Some(ref params) = eatmy {
+            let eatmy = EMDParams::new(
+                self.codename.as_ref().unwrap(), &self.config.arch
+            );
+            if let Some(params) = eatmy {
                 if params.needs_universe {
                     debug!("Add Universe for eat my data");
                     self.enable_universe()?;
                     self.add_universe(ctx)?;
                 }
+
                 apt_get_update::<&str>(ctx, &[], &mut self.apt_update)?;
-                apt_get_install(ctx, &[params.package], &EatMyData::No)?;
-                self.eatmydata = EatMyData::Preload(params.preload);
-            } else {
-                info!("Unsupported distribution for eatmydata. Ignoring");
-                self.eatmydata = EatMyData::No;
-            }
-        }
-        Ok(())
-    }
-    #[cfg(feature="containers")]
-    fn ensure_locales(&mut self, ctx: &mut Context)
-        -> Result<(), StepError>
-    {
-        if self.eatmydata == EatMyData::Need {
-            self.ensure_codename(ctx)?;
-            let eatmy = EMDParams::find(
-                self.codename.as_ref().unwrap(), &self.config.arch);
-            if let Some(ref params) = eatmy {
-                if params.needs_universe {
-                    debug!("Add Universe for eat my data");
-                    self.enable_universe()?;
-                    self.add_universe(ctx)?;
+                if let Ok(()) = apt_get_install(ctx, &[params.package], &EatMyData::No) {
+                    self.eatmydata = params.find(ctx);
+                } else {
+                    warn!("Could not install {} package", params.package);
```

Comment:
Can we also emit error here?

Reply:
I'm afraid in future ubuntu releases eatmydata package may be renamed.

Reply:
So in this way we won't break future ubuntu versions.

Reply:
Oh, you mean `error!` )))

Reply:
No I mean:
```
 match apt-get-install... {
 Err(e) => warn!("some intro {:#}", e), // << so we see actual error here
 }
```

Reply:
I think it does not add any new information (we say clearly which package cannot be installed):
```
Could not install libeatmydata1 package. The build may be slower than usual: error running <Command "/usr/bin/apt-get" "-oDir::cache::pkgc
ache=" "-oDir::cache::srcpkgcache=" "install" "-y" "libeatmydata1-asdf"; environ[5]; chroot="/vagga/root"; work-dir="/work"> exited with code 100
```

Reply:
Yes, but this adds information:
1. Whether the apt-get command could run (i.e. /usr/bin/apt-get is executable)
2. Whether all the respective volumes could be mounted
etc.

There are so many things that could go wrong beyond non-existence of the package. And if everything is okay, we won't get any message. So it's better to add more info if it fails.